pkg/grpc/universalapi/api_metadata_test.go

```diff
@@ -58,40 +61,89 @@ func TestGetMetadata(t *testing.T) {
-			return capsMap
+	testcases := []struct {
+		name                     string
+		HealthCheck              *apphealth.Config
+		expectHealthCheckEnabled bool
+	}{
+		{
+			name: "health check enabled",
+			HealthCheck: &apphealth.Config{
+				ProbeOnly:     true,
+				ProbeInterval: 10 * time.Second,
+				ProbeTimeout:  5 * time.Second,
+				Threshold:     3,
+			},
+			expectHealthCheckEnabled: true,
 		},
-		ExtendedMetadata: map[string]string{
-			"testKey": "testValue",
+		{
+			name:                     "health check disabled",
+			HealthCheck:              nil,
+			expectHealthCheckEnabled: false,
 		},
 	}
 
-	response, err := fakeAPI.GetMetadata(context.Background(), &emptypb.Empty{})
-	require.NoError(t, err, "Expected no error")
-	assert.Equal(t, response.Id, "fakeAPI")
-	assert.Len(t, response.RegisteredComponents, 1, "One component should be returned")
-	assert.Equal(t, response.RegisteredComponents[0].Name, "testComponent")
-	assert.Contains(t, response.ExtendedMetadata, "testKey")
-	assert.Equal(t, response.ExtendedMetadata["testKey"], "testValue")
-	assert.Equal(t, response.ExtendedMetadata[daprRuntimeVersionKey], buildinfo.Version())
-	assert.Len(t, response.RegisteredComponents[0].Capabilities, 1, "One capabilities should be returned")
-	assert.Equal(t, response.RegisteredComponents[0].Capabilities[0], "mock.feat.testComponent")
-	assert.Equal(t, response.GetActiveActorsCount()[0].Type, "abcd")
-	assert.Equal(t, response.GetActiveActorsCount()[0].Count, int32(10))
-	assert.Len(t, response.Subscriptions, 1)
-	assert.Equal(t, response.Subscriptions[0].PubsubName, "test")
-	assert.Equal(t, response.Subscriptions[0].Topic, "topic")
-	assert.Equal(t, response.Subscriptions[0].DeadLetterTopic, "dead")
-	assert.Equal(t, response.Subscriptions[0].PubsubName, "test")
-	assert.Len(t, response.Subscriptions[0].Rules.Rules, 1)
-	assert.Equal(t, fmt.Sprintf("%s", response.Subscriptions[0].Rules.Rules[0].Match), "")
-	assert.Equal(t, response.Subscriptions[0].Rules.Rules[0].Path, "path")
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			appConnectionConfig := config.AppConnectionConfig{
+				Port:                1234,
+				Protocol:            "http",
+				ChannelAddress:      "1.2.3.4",
+				MaxConcurrency:      10,
+				HealthCheckHTTPPath: "/healthz",
+				HealthCheck:         tc.HealthCheck,
+			}
+
+			fakeAPI := &UniversalAPI{
+				AppID:     "fakeAPI",
+				Actors:    mockActors,
+				CompStore: compStore,
+				GetComponentsCapabilitesFn: func() map[string][]string {
+					capsMap := make(map[string][]string)
+					capsMap["testComponent"] = []string{"mock.feat.testComponent"}
+					return capsMap
+				},
+				ExtendedMetadata: map[string]string{
+					"testKey": "testValue",
+				},
+				AppConnectionConfig: appConnectionConfig,
+			}
+
+			response, err := fakeAPI.GetMetadata(context.Background(), &emptypb.Empty{})
+			require.NoError(t, err, "Expected no error")
+			assert.Equal(t, response.Id, "fakeAPI")
```

Comment:
Just a thought... Consider serializing this object as JSON and comparing the JSON with the response? This is getting very complex to follow, with too many asserts one for each property.

Reply:
Sounds good, refactored!